specialisation/homework1.py

```diff
@@ -0,0 +1,38 @@
+import abc
+class Shape(object):
+    __metaclass__ = abc.ABCMeta\
+
+    @abc.abstractmethod
+    def calc_perimeter(self, input):
+        """Method documentation"""
+        return
+
+class Triangle(Shape):
+    def __init__(self, a, b, c):
+        self.a = a
+        self.b = b
+        self.c = 
```

Comment:
`self.c ` is not assigned with any value, this will produce a compilation error.